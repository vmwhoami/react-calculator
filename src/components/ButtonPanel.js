import React from 'react';
import Button from './Button';

const ButtonPanel = () => {
  const group1 = ['AC', '+/-', '%', 'รท'];
  const group2 = ['7', '8', '9', 'X'];
  const group3 = ['4', '5', '6', '-'];
  const group4 = ['1', '2', '3', '+'];
  const group5 = ['0', '.', '='];

  return (
    <>
      <div className="group1">
        {group1.map(element => <Button name={element} key={`${element}`} />)}
      </div>
      <div className="group2">
        {group2.map(element => <Button name={element} key={`${element}`} />)}
      </div>
      <div className="group3">
        {group3.map(element => <Button name={element} key={`${element}`} />)}
      </div>
      <div className="group4">
        {group4.map(element => <Button name={element} key={`${element}`} />)}
      </div>
      <div className="group5">
        {group5.map(element => <Button name={element} key={`${element}`} />)}
      </div>
    </>
  );
};

export default ButtonPanel;
