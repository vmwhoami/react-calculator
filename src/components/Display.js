import React from 'react';
import PropTypes from 'prop-types';

const Display = props => {
  const { next, result } = props;
  return (
    <>
      <h2>
        {next || result || '0'}
      </h2>
    </>
  );
};

Display.defaultProps = {
  result: '0',
  next: null,
};
Display.propTypes = {
  result: PropTypes.string,
  next: PropTypes.string,
};

export default Display;
