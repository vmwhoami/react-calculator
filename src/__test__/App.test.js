import React from 'react';
import renderer from 'react-test-renderer';
import Display from '../components/Display';
import ButtonPanel from '../components/ButtonPanel';

it('renders quote page correctly', () => {
  const tree = renderer
    .create(
      <div className="calculator">
        <Display />
        <ButtonPanel />
      </div>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
