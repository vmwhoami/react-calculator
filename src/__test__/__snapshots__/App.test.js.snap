// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders quote page correctly 1`] = `
<div
  className="calculator"
>
  <h2
    className="display"
  >
    0
  </h2>
  <div
    className="groups"
  >
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="button orange"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="groups"
  >
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="button orange"
      onClick={[Function]}
      type="button"
    >
      X
    </button>
  </div>
  <div
    className="groups"
  >
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="button orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="groups"
  >
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="button orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="groups"
  >
    <button
      className="button wide"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="button orange"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
